### DISCUSSION TITLE
Flowise + elevenlabs (text to speech)

### DISCUSSION THREAD

**Discussion by Warcylewis** [FlowiseAI/Flowise#1353] [Q&A]
Hey community,
I would love to get your help on this or thoughts on how potential solve my issue:
**Goal:**
I want to create a flowise chatbot that can also read his own message out loud with elevenlabs  (https://www.npmjs.com/package/elevenlabs-node)

**What I have done so far:**
-I installed the elevenlabs node in flowise/packages/components  like any external node dependency should be in the project (at least what I saw on other youtube videos)
-this created in my package.json""elevenlabs-node": "^2.0.1"
-I then yarn install & build & start to go for local:3000 -> created a custome tool 
-> ran it "create a voice message of XYZ"

**Problem:**
"can't find module 'elevenlabs-node"

**QUESTION:**
what am I doing wrong or potential have forgotten to do?

**Output:**
[agent/action] [1:chain:AgentExecutor] Agent selected action: {
  "tool": "text_to_speech_stream",
  "toolInput": {
    "t2sMessage": "Hey Stranger, you are amazing and you rock and thanks for your help!"
  },
  "log": "Invoking \"text_to_speech_stream\" with {\n  \"t2sMessage\": \"Hey Stranger, you are amazing and you rock and thanks for your help!\"\n}\n",
  "messageLog": [
    {
      "lc": 1,
      "type": "constructor",
      "id": [
        "langchain",
        "schema",
        "AIMessageChunk"
      ],
      "kwargs": {
        "content": "",
        "additional_kwargs": {
          "function_call": {
            "name": "text_to_speech_stream",
            "arguments": "{\n  \"t2sMessage\": \"Hey Stranger, you are amazing and you rock and thanks for your help!\"\n}"
          }
        }
      }
    }
  ]
}
[tool/start] [1:chain:AgentExecutor > 3:tool:DynamicStructuredTool] Entering Tool run with input: "{"t2sMessage":"Hey Stranger, you are amazing and you rock and thanks for your help!"}"
[tool/error] [1:chain:AgentExecutor > 3:tool:DynamicStructuredTool] [13ms] Tool run errored with error: "Cannot find module 'elevenlabs-node'"
[llm/start] [1:chain:AgentExecutor > 4:llm:ChatOpenAI] Entering LLM run with input: {
  "messages": [......

**Comment by HenryHengZJ** [comment]
@Warcylewis you have to specify the library in the .env variable - `TOOL_FUNCTION_EXTERNAL_DEP`
https://docs.flowiseai.com/integrations/tools/custom-tool#import-external-dependencies

**Comment by davide445** [comment]
@Warcylewis can you please share what video tutorial have you followed on this topic.

**Comment by kimarawatson** [comment]
It looks like the issue is that the elevenlabs-node package isn’t installed where the Flowise runtime can find it. Try installing it at the root of the Flowise project with yarn add elevenlabs-node, then run yarn build and restart the server. Also, double-check your import statements to make sure you’re importing the module correctly. Let me know if you want a quick example of how to integrate it in a custom tool!

**Reply by MaskedLlama04** [reply]
Hey there @PavloSavvy , would you have some JSON of this kind of project perhaps? I'm really trying to do what @Warcylewis , but I can't find anything on the interent and I'm a bit lost.

**Comment by MaskedLlama04** [comment]
Hey there @Warcylewis could you please share with me the final workflow you achieved? I really need some help with the kind of project you wanted to do
